test/integration/config/index.tap.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const Config = require('../../../lib/config')
+const tap = require('tap')
+
+tap.test('Configuration Library', function (harness) {
+  harness.autoend()
+
+  let originalConsoleError
+
+  harness.beforeEach(function () {
+    // Make sure we don't pollute our logs
+    originalConsoleError = global.console.error
+    global.console.error = function () {}
+  })
+
+  harness.afterEach(function () {
+    // Restore so we don't have a knock-on effect with other test suites
+    global.console.error = originalConsoleError
+  })
+
+  harness.test('it should gracefully handle errors on instantiation', function (test) {
+    test.plan(2)
```

Comment:
I used to also count the number of tests and put that count inside the plan, but I've now favoured doing `t.end()` at the end of each subtest. It's a bit more verbose but it also has the advantage that it shows you when a test is intended to end. Plus, no need to manually count the number of tests, which gets annoying when you add or delete tests. I've also caught some bugs in my test that way when asynchronous execution didn't quite end my tests in the way I intended them to.